### PR TITLE
travis: remove .travis.yml and build scripts

### PR DESCRIPTION
**Pull request by cgundogan**
No need to keep travis-related scripts and configuration files in the `RIOT` directory around. They are still retrievable via git's history.

@kaspar030 murdock is not using these scripts, right?

### CODE REVIEW

**Comment by miri64** [comment]
1. Needs rebase
2. `dist/tools/travis-scripts` *is* used AFAIK.

**Comment by cgundogan** [comment]
>1. Needs rebase

Why?

**Comment by kaspar030** [comment]
> dist/tools/travis-scripts is used AFAIK.

yep it is...

**Comment by cgundogan** [comment]
> dist/tools/travis-scripts is used AFAIK.
>> yep it is...

@kaspar030 which script is murdock using? we could move that to a `murdock` folder. I looked at [1], but couldn't find a hint.

[1] https://github.com/kaspar030/murdock

**Comment by smlng** [comment]
Maybe some other use Travis on their RIOT fork and require/want that file to remain in place? Further, we also keep the Drone config by @gebart around and it's not used by RIOT itself.

**Comment by miri64** [comment]
>> Needs rebase
>
> Why?

Because I'm not used to the new GitHub feature yet 😅. Merge-block looks like merge-conflict used to look.

**Comment by miri64** [comment]
Mh, given @smlng's comment, I would suggest closing this PR...

**Comment by cgundogan** [comment]
>Mh, given @smlng's comment, I would suggest closing this PR...

I don't think that anybody feels responsible to maintain our travis scripts (beyond what @kaspar030 needs for Murdock). So I am still against keeping (from RIOT's perspective) unnecessary configuration files. Anyon else who forks RIOT and wants to use such features, can provide such files in her/his fork.

**Comment by smlng** [comment]
CI is a community thing, so we may need to discuss this on the devel mailing list? Making a decision within a circle of 4, might not be the wisest thing to do here.

If somebody uses travis --and/or is willing to maintain required scripts-- we can (and should) keep the travis config. And again, we have this `.drone.yml`, too ...

**Comment by BytesGalore** [comment]
Taking the comments [1] of @gebart on the devel list into account, I propose to close this PR.

[1] https://lists.riot-os.org/pipermail/devel/2016-September/004607.html

**Comment by miri64** [comment]
Second proposal to close this PR. I set the contested tag ;-).

**Comment by cgundogan** [comment]
I demand that anyone who wants to close this PR should first check that our `.travis.yml` configuration actually still works with the current travis server (; Otherwise, I don't see the point to keep it around.

**Comment by miri64** [comment]
To be honest I don't have any problem with just removing `.travis.yml` (which for some reason is conflicting btw), but the other files are required by Murdock, as stated before.

**Comment by cgundogan** [comment]
it's only `dist/tools/travis-scripts/build_and_test.sh` that is required by murdock and drone. I can move the file into a folder with a more general name (`ci_scripts`) and change drone to point to that. Murdock however would need some adaptation from @kaspar030's side. I will ping once I propose the change

**Comment by kaspar030** [comment]
> it's only dist/tools/travis-scripts/build_and_test.sh that is required by murdock and drone. I can move the file into a folder with a more general name (ci_scripts) and change drone to point to that.

I think it's a good idea to rename 'travis-scripts' into something more general (I'd prefer just 'ci').

IMO removing travis.yaml is fine, as travis was already choking back then when our build matrix was half of its current size, and probably noone is using it anyways.

File: dist/tools/travis-scripts/build_and_test.sh [hunk 1]
```diff
@@ -1,89 +0,0 @@
-#!/usr/bin/env bash
```
Comment (kaspar030) [inline dist/tools/travis-scripts/build_and_test.sh]: This file is needed by murdock.

**Comment by cgundogan** [comment]
@kaspar030 could you have look at the changes in `dist/tools/ci/build_and_test.sh `?

**Comment by kaspar030** [comment]
> @kaspar030 could you have look at the changes in dist/tools/ci/build_and_test.sh?

They look fine to me!

Please rebase.
@miri64 you still think this PR is contested?

**Comment by miri64** [comment]
Nope.

**Comment by smlng** [comment]
@cgundogan please adapt `.drone.yml` with the new script path. 

I'm still against removing `.travis.yml`, however, if majority is okay with it: proceed.

**Comment by miri64** [comment]
@smlng in the end it is a question of maintainability. If e.g. you would ensure that the `.travis.yml` would keep working (e.g. by building master regularly on your repo using Travis), I wouldn't be opposed to not removing the file.

**Comment by cgundogan** [comment]
>@cgundogan please adapt .drone.yml with the new script path.

@smlng what do you mean by that? I already did that or did I miss something?

**Comment by smlng** [comment]
@cgundogan oops, my bad.

**Comment by cgundogan** [comment]
so this PR is basically merge-ready? Probably @kaspar030 should merge this, so that murdock does not go out-of-sync

**Comment by smlng** [comment]
@cgundogan ACK, and yes @kaspar030 may have the honor to merge this one 😄 Maybe I'll have a look into travis later, then I'll re-add its config, but for the time being I can live without it.

**Comment by OlegHahm** [comment]
@kaspar030, are you aware of this?

**Comment by kaspar030** [comment]
> @kaspar030, are you aware of this?

Yes. I'm waiting for a few murdock changes to pile up, and for free time to update it. I'll merge then.

**Comment by kaspar030** [comment]
Murdock's scripts are adapted. go!

**Pull request merged**